src/base/object_representation.h

```diff
@@ -601,9 +601,20 @@
 #define OR_REP_OFFSET    0
 #define OR_MVCC_REP_SIZE 4
 
-/* CHN fixed-offset in NON-MVCC only, in MVCC the CHN offset is variable */
-#define OR_NON_MVCC_CHN_OFFSET   4
+#define OR_MVCC_FLAG_OFFSET OR_REP_OFFSET
+#define OR_MVCC_FLAG_SIZE OR_MVCC_REP_SIZE
 
+#define OR_CHN_OFFSET (OR_REP_OFFSET + OR_MVCC_REP_SIZE)
+#define OR_CHN_SIZE 4
+
+#define OR_MVCC_INSERT_ID_OFFSET (OR_CHN_OFFSET + OR_CHN_SIZE)
+#define OR_MVCC_INSERT_ID_SIZE 8
+
+#define OR_MVCC_DELETE_ID_OFFSET (OR_MVCC_INSERT_ID_OFFSET + OR_MVCC_INSERT_ID_SIZE)
```

Comment:
`OR_MVCC_DELETE_ID_OFFSET` is a bit unclear. It is declared as fixed, but actually it depends on mvcc flags unlike `OR_MVCC_INSERT_ID_OFFSET.` 

It would be more simple to check the flags in this macro and use it freely in other occurences.
You could change it as `#define OR_MVCC_DELETE_ID_OFFSET(mvcc_flags) (OR_MVCC_INSERT_ID_OFFSET + (((mvcc_flags) & OR_MVCC_FLAG_VALID_INSID)  ? OR_MVCC_INSERT_ID_SIZE : 0))` and simplify  `OR_GET_MVCC_DELETE_ID` also.

`OR_MVCC_PREV_VERSION_LSA_OFFSET` would need the same change.